EncoreTickets.SDK/Utilities/Common/Serializers/JsonSerializer.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.IO;
+using Newtonsoft.Json;
+using RestSharp;
+using RestSharp.Deserializers;
+using RestSharp.Serializers;
+
+namespace EncoreTickets.SDK.Utilities.Common.Serializers
+{
+    abstract class JsonSerializer : ISerializer, IDeserializer
+    {
+        protected abstract Newtonsoft.Json.JsonSerializer Serializer { get; }
+
+        public static T CreateInstance<T>(string dateFormat = null) 
+            where T : JsonSerializer, new()
+        {
+            var serializer = new T();
+            if (!string.IsNullOrEmpty(dateFormat))
+            {
+                serializer.Serializer.DateFormatString = dateFormat;
+            }
```

Comment:
maybe it’s better to save and configure here not the full serializer, but the settings for the serializer (JsonSerializerSettings)?